src/test/common/commandExecutor.test.ts

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for details.
+
+import {CommandExecutor} from "../../common/commandExecutor";
+import {Log} from "../../common/log/log";
+
+import * as assert from "assert";
+import * as semver from "semver";
+import * as sinon from "sinon";
+
+suite("commandExecutor", function() {
+    suite("commonContext", function() {
+        teardown(function() {
+            let mockedMethods = [Log.logMessage, Log.logCommandStatus];
+
+            mockedMethods.forEach((method) => {
+                if (method.hasOwnProperty("restore")) {
+                    (<any>method).restore();
+                }
+            });
+        });
+
+        test("should execute a command", function() {
+            let ce = new CommandExecutor();
+            let actualExecuteResult: string[] = [];
+
+            sinon.stub(Log, "logMessage", function(message: string) {
+                actualExecuteResult.push(message.replace("\n", ""));
+                console.log(message);
+            });
+
+            return ce.execute("node -v")
```

Comment:
Printing the executing and finished text is an important part of the commandExecutor class, so I'd verify that those things are printed properly. executing echo 23423432 or a command that returns something constant might make that easier. Ideally we should also test that both stderr and stdout of the command are printed properly.